### PR TITLE
TT-196

### PR DESCRIPTION
**Pull request by MontgomeryWatts**
Acts like a SPA

### CODE REVIEW

File: src/components/CommitteesMenu.vue [hunk 1]
```diff
@@ -42,7 +42,6 @@ export default {
   methods: {
     redirect (itemId) {
       this.$router.push('/committee/' + itemId)
-      this.$router.go()
```
Comment (MontgomeryWatts) [inline src/components/CommitteesMenu.vue]: Page navigation happens without this.

File: src/pages/Committee.vue [hunk 2]
```diff
@@ -139,15 +148,6 @@ export default {
       username: 'username',
       admin: 'admin'
     })
-  },
-  /* Since this component is used for each committee page, we have to
-    watch for changes in the URL and update the props on the page
-    when the route changes */
-  watch: {
-    '$route.params.committee': function (committee) {
```
Comment (MontgomeryWatts) [inline src/pages/Committee.vue]: The logic here is inconsistent with the logic above in `beforeMount` and I'm not sure if the watch actually worked.

File: src/pages/Committee.vue [hunk 3]
```diff
@@ -107,17 +107,26 @@ export default {
...
-        token: token,
-        committee_id: this.$router.history.current.params['committee']
+    this.updatePage()
+  },
+  beforeRouteUpdate (to, from, next) {
+    this.updatePage()
+    next()
+  },
+  methods: {
+    updatePage () {
```
Comment (MontgomeryWatts) [inline src/pages/Committee.vue]: This function is taken wholly from the old `beforeMount` function.

File: src/pages/Committee.vue [hunk 4]
```diff
@@ -107,17 +107,26 @@ export default {
...
-        committee_id: this.$router.history.current.params['committee']
-      })
-      this.$socket.emit('get_minutes', {
-        token: token,
-        committee_id: this.$router.history.current.params['committee']
+    this.updatePage()
+  },
+  beforeRouteUpdate (to, from, next) {
+    this.updatePage()
+    next()
```
Comment (MontgomeryWatts) [inline src/pages/Committee.vue]: This lets the page properly refresh when navigating to a new committee when already on a different committee's page. You can read more about the Vue router guards here: https://router.vuejs.org/guide/advanced/navigation-guards.html#in-component-guards

File: src/pages/Committee.vue (hunk 2 above)
Comment (lukasyelle) [reply src/pages/Committee.vue]: Have you noticed any issues switching committees? I see that those events you got rid of didn't include a token but the ones in beforeMount did, so they may have never worked in the first place.

File: src/pages/Committee.vue (hunk 4 above)
Comment (elijah-parrish) [reply src/pages/Committee.vue]: Great resource there.

**Pull request merged**